File: core/src/com/github/emulio/utils/Tail.java
package com.github.emulio.utils;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

/**
 * Java implementation of the Unix tail command
 * 
 * @param args[0]
 *            File name
 * @param args[1]
 *            Update time (seconds). Optional. Default value is 1 second
 * 
 * @author Luigi Viggiano (original author)
 *         http://it.newinstance.it/2005/11/19/listening-changes-on-a-text-file-
 *         unix-tail-implementation-with-java/
 * @author Alessandro Melandri (modified by)
 */
public class Tail {

//	static long sleepTime = 1000;

//	public static void main(String[] args) throws IOException {
//
//		if (args.length > 0) {
//
//			if (args.length > 1)
//				sleepTime = Long.parseLong(args[1]) * 1000;
//
//			String fileName = args[0];
//			File file = new File(fileName);
//
//			tailFile(file);
//
//		} else {
//			System.out.println(
//					"Missing parameter!\nUsage: java JavaTail fileName [updateTime (Seconds. default to 1 second)]");
//		}
//	}

	public static void tailFile(final File file, final long sleepTime) throws FileNotFoundException, IOException {
		BufferedReader input = new BufferedReader(new FileReader(file));
		String currentLine = null;

		while (true) {

			if ((currentLine = input.readLine()) != null) {
				System.out.println(currentLine);
				continue;
			}

			try {
				Thread.sleep(sleepTime);
			} catch (InterruptedException e) {
				Thread.currentThread().interrupt();
				break;
			}

		}
		input.close();
	}
}

File: core/src/com/github/emulio/utils/Tail.kt
package com.github.emulio.utils

import java.io.BufferedReader
import java.io.File
import java.io.FileNotFoundException
import java.io.FileReader
import java.io.IOException

/**
 * Java implementation of the Unix tail command
 *
 * @param args[0]
 * File name
 * @param args[1]
 * Update time (seconds). Optional. Default value is 1 second
 *
 * @author Luigi Viggiano (original author)
 * http://it.newinstance.it/2005/11/19/listening-changes-on-a-text-file-
 * unix-tail-implementation-with-java/
 * @author Alessandro Melandri (modified by)
 */
class Tail {

//    internal var sleepTime: Long = 1000
//
//    @Throws(IOException::class)
//    @JvmStatic
//    fun main(args: Array<String>) {
//
//        if (args.size > 0) {
//
//            if (args.size > 1)
//                sleepTime = java.lang.Long.parseLong(args[1]) * 1000
//
//            val fileName = args[0]
//            val file = File(fileName)
//
//            tailFile(file)
//
//        } else {
//            println(
//                    "Missing parameter!\nUsage: java JavaTail fileName [updateTime (Seconds. default to 1 second)]")
//        }
//    }

    @Throws(FileNotFoundException::class, IOException::class)
    fun tailFile(file: File, sleepTime: Long) {
        val input = BufferedReader(FileReader(file))

        var currentLine: String? = input.readLine()
        while (true) {

            if (currentLine != null) {
                println(currentLine)
                currentLine = input.readLine()
                continue
            }

            try {
                Thread.sleep(sleepTime)
            } catch (e: InterruptedException) {
                Thread.currentThread().interrupt()
                break
            }

        }
        input.close()
    }
}